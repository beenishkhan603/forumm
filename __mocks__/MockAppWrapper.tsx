import { AuthContext } from '@libs/useAuth'
import { MockedProvider, MockedResponse } from '@apollo/client/testing'
import { ProfileInfo } from '@type/Hooks/useAuth/Auth.type'
import ALL_MOCKS from './graphql/mocks'

export const defaultMockProfile: ProfileInfo = {
  company: '448 Studio',
  email: 'test@email.com',
  fullName: 'John Doe',
  university: '448 Studio',
  profileImageUrl: '/test.png',
  userId: '123',
  groups: [],
}

const MockAppWrapper = ({
  children,
  mocks,
  mockProfile = defaultMockProfile,
}: {
  mockProfile?: Partial<ProfileInfo>
  children: any
  mocks?: readonly MockedResponse<Record<string, any>>[] | undefined
}) => {
  const Layout = children?.type?.Layout ?? null

  const GQL_MOCKS: MockedResponse[] = ALL_MOCKS

  if (mocks) mocks.forEach((m) => GQL_MOCKS.push(m))
  return (
    <MockedProvider
      mocks={GQL_MOCKS}
      defaultOptions={{
        query: {
          errorPolicy: 'all',
          fetchPolicy: 'cache-first',
        },
        watchQuery: {
          errorPolicy: 'all',
          fetchPolicy: 'cache-and-network',
        },
        mutate: {
          errorPolicy: 'all',
        },
      }}
    >
      <AuthContext.Provider
        value={{
          profile: mockProfile
            ? { ...defaultMockProfile, ...mockProfile }
            : defaultMockProfile,
          setProfile: () => {},
        }}
      >
        <div id="modal-root" />
        {Layout ? <Layout>{children}</Layout> : children}
      </AuthContext.Provider>
    </MockedProvider>
  )
}

export default MockAppWrapper
