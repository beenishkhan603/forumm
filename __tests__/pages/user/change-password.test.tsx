import { render } from '@testing-library/react'
import ChangePasswordPage from 'pages/user/change-password'
import MockAppWrapper from '__mocks__/MockAppWrapper'

it('renders change password page unchanged', () => {
  const { container } = render(
    <MockAppWrapper>
      <ChangePasswordPage />
    </MockAppWrapper>
  )
  expect(container).toMatchSnapshot()
})
