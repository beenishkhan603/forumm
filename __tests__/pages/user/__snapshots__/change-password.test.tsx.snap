// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders change password page unchanged 1`] = `
<div>
  <div
    id="modal-root"
  />
  <div
    class="flex justify-center"
    style="color: rgb(24, 26, 32);"
  >
    <div
      class="w-full md:w-1/2 my-16 relative"
      style="color: rgb(24, 26, 32);"
    >
      <div
        class="absolute transition-all !text-white p-8 top-0 bottom-0 left-0 right-0 flex items-center justify-center h-24 mt-auto bg-green-500 pointer-events-none opacity-0 rounded text-center"
        style="color: rgb(24, 26, 32);"
      >
        Your password has been successfully updated!
        <svg
          class="inline-block h-8 w-8 ml-2"
        />
      </div>
      <div
        class="text-white text-center text-2xl font-bold pl-3 md:pl-0"
        style="color: rgb(24, 26, 32);"
      >
        Change Password
      </div>
      <form
        class=""
      >
        <div
          class="relative"
          style="color: rgb(24, 26, 32);"
        >
          <label
            class="flex flex-col my-4 "
          >
            <span
              class="absolute text-sm mb-2 ml-2 z-10 pl-2 transition-all  !backdrop-blur-md"
              style="transform: translateY(-50%);"
            >
              <div
                class="flex flex-col"
                style="color: rgb(24, 26, 32);"
              >
                <span
                  class="z-10  after:content-['*'] after:mx-2 after:text-red-500"
                >
                  New Password
                </span>
              </div>
            </span>
            <input
              autocomplete="New Password"
              class="rounded-2xl transition-all bg-transparent border  rounded px-4 py-4 text-sm outline-none placeholder-gray-500"
              id="New Password"
              placeholder="*******"
              required=""
              type="password"
              value=""
            />
          </label>
          <div
            class="absolute right-3 top-[15px] text-xl items-center"
            style="color: rgb(24, 26, 32);"
            type="button"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0zm0 0h24v24H0zm0 0h24v24H0zm0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7zM2 4.27l2.28 2.28.46.46A11.804 11.804 0 001 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3 2 4.27zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2zm4.31-.78l3.15 3.15.02-.16c0-1.66-1.34-3-3-3l-.17.01z"
              />
            </svg>
          </div>
        </div>
        <div
          class="relative"
          style="color: rgb(24, 26, 32);"
        >
          <label
            class="flex flex-col my-4 "
          >
            <span
              class="absolute text-sm mb-2 ml-2 z-10 pl-2 transition-all  !backdrop-blur-md"
              style="transform: translateY(-50%);"
            >
              <div
                class="flex flex-col"
                style="color: rgb(24, 26, 32);"
              >
                <span
                  class="z-10  after:content-['*'] after:mx-2 after:text-red-500"
                >
                  Confirm New Password
                </span>
              </div>
            </span>
            <input
              autocomplete="Confirm New Password"
              class="rounded-2xl transition-all bg-transparent border  rounded px-4 py-4 text-sm outline-none placeholder-gray-500"
              id="Confirm New Password"
              placeholder="Please retype your new password"
              required=""
              type="password"
              value=""
            />
          </label>
          <div
            class="absolute right-3 top-[15px] text-xl items-center"
            style="color: rgb(24, 26, 32);"
            type="button"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0zm0 0h24v24H0zm0 0h24v24H0zm0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7zM2 4.27l2.28 2.28.46.46A11.804 11.804 0 001 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3 2 4.27zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2zm4.31-.78l3.15 3.15.02-.16c0-1.66-1.34-3-3-3l-.17.01z"
              />
            </svg>
          </div>
        </div>
        <button
          class="
      bg-forumm-blue-2 text-white relative rounded-full  px-6 py-4 text-base h-8 flex items-center hover:animate-heartbeat justify-center transition-all w-full my-8 "
          style="background-color: rgb(55, 99, 233);"
          type="submit"
        >
          <span
            class="relative z-10"
            style="color: white;"
          >
            Change Password
          </span>
        </button>
      </form>
    </div>
  </div>
</div>
`;
